File: modules/examples/src/material/demo_common.dart
library angular2_examples.material.demo_common;

import 'package:angular2/src/core/dom/browser_adapter.dart';
import 'package:angular2/src/core/services/url_resolver.dart';

void commonDemoSetup() {
  BrowserDomAdapter.makeCurrent();
}

class DemoUrlResolver extends UrlResolver {
  @override
  String resolve(String baseUrl, String url) {
    const MATERIAL_PKG = 'package:angular2_material/';

    // We run a proxy server in front of pub serve that prepends "example" to
    // paths
    if (url.startsWith(MATERIAL_PKG)) {
      return '/examples/packages/angular2_material/' +
          url.substring(MATERIAL_PKG.length);
    }
    return super.resolve(baseUrl, url);
  }
}

File: modules/playground/src/material/demo_common.ts
import {print} from 'angular2/src/facade/lang';
import {UrlResolver} from 'angular2/src/compiler/url_resolver';
import {isPresent, isBlank, RegExpWrapper, StringWrapper} from 'angular2/src/facade/lang';
import {DOM} from 'angular2/src/core/dom/dom_adapter';
import {Injectable} from 'angular2/core';
import {BrowserDomAdapter} from 'angular2/src/core/dom/browser_adapter';


export function commonDemoSetup(): void {
  BrowserDomAdapter.makeCurrent();
}

@Injectable()
export class DemoUrlResolver extends UrlResolver {
  constructor() {
    super();
  }

  resolve(baseUrl: string, url: string): string {
    // The standard UrlResolver looks for "package:" templateUrls in
    // node_modules, however in our repo we host material widgets at the root.
    if (url.startsWith('package:angular2_material/')) {
      return '/' + url.substring(8);
    }
    return super.resolve(baseUrl, url);
  }
}
